### PR TITLE
Upload (daily) CI results to Hub

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Let's start to upload the results to a Hub dataset [hf-internal-testing/transformers_daily_ci](https://huggingface.co/datasets/hf-internal-testing/transformers_daily_ci/tree/main/2024-05-31) and use this dataset to build a visualization dashboard 🚀 !

So far I take the advantage of the daily CI workflow run - upload directly from there instead of fetching the finished workflow run afterward.

I didn't change any format in the raw reports (but processed in the workflow), but I believe it contains everything required to perform visualization/analysis.

[A dummy run](https://github.com/huggingface/transformers/actions/runs/9320789057) gives [hf-internal-testing/transformers_daily_ci](https://huggingface.co/datasets/hf-internal-testing/transformers_daily_ci/tree/main/2024-05-31)

### CODE REVIEW

File: .github/workflows/slack-report.yml
Comment (ydshieh) [inline .github/workflows/slack-report.yml:23]: This is a new fine grained token with write access to [hf-internal-testing/transformers_daily_ci](https://huggingface.co/datasets/hf-internal-testing/transformers_daily_ci)

File: utils/notification_service.py
Comment (ydshieh) [inline utils/notification_service.py:1161]: Only upload the results for the runs triggered by the  scheduled event .

File: .github/workflows/slack-report.yml
Comment (LysandreJik) [reply .github/workflows/slack-report.yml:23]: Cool!

**Review by LysandreJik — APPROVED** [review]
Awesome!

**Comment by ydshieh** [comment]
Thanks for the review. I will merge once I applied the same changes to

> utils/notification_service_quantization.py

(sorry, I forget this file quite frequently)

**Pull request merged**
